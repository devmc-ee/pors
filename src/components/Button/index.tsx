import { SyntheticEvent } from 'react';

export interface IButton {
    classes?: string;
    value: number;
    onClick: (event: SyntheticEvent) => void;
    children: JSX.Element | string;
}
const Button = ({ classes = '', children, onClick, value = 0 }: IButton) => {
    return (
        <button
            className={'waves-effect waves-light btn-large btn-menu ' + classes}
            data-value={value}
            onClick={onClick}
        >
            {children}
        </button>
    );
};
export default Button;
