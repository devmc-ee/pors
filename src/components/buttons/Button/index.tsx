import { SyntheticEvent } from 'react';

export interface IButton {
    classes?: string;
    value: number | string;
    onClick: (event: SyntheticEvent) => void;
    children: JSX.Element | string;
}
const Button = ({ classes = '', children, onClick, value = 0 }: IButton) => {
    return (
        <button className={'button ' + classes} data-value={value} onClick={onClick}>
            {children}
        </button>
    );
};
export default Button;
