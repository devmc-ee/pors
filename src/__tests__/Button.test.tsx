import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Button, { IButton } from '../components/shared/buttons/Button';

const setUp = (props: IButton) => render(<Button {...props} />);

afterEach(cleanup);

const handleClick = jest.fn();

const buttonProps = {
    classes: ' btn-test-class',
    onClick: handleClick,
    children: 'TestMe',
    value: 100,
};
describe('Should render Button component', () => {
    it('has all props values', () => {
        const { getByRole } = setUp(buttonProps);
        const btn = getByRole('button');
        expect(btn).toHaveClass('btn-test-class');
        expect(btn).toHaveTextContent('TestMe');
    });

    it('calls function on click 2 times', () => {
        const { getByRole } = setUp(buttonProps);

        fireEvent.click(getByRole('button'));
        fireEvent.click(getByRole('button'));
        expect(handleClick).toHaveBeenCalledTimes(2);
    });
    it('has  data-value attribute', () => {
        const { getByRole } = setUp(buttonProps);
        const btn = getByRole('button');
        expect(btn).toHaveAttribute('data-value');
    });

    it('matches the snapshot with value = 200', () => {
        buttonProps.value = 200;
        const tree = renderer.create(<Button {...buttonProps} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
