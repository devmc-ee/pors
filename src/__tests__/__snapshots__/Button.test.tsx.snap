// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Button component matches the snapshot with value = 200 1`] = `
<button
  className="button  btn-test-class"
  data-value={200}
  onClick={[MockFunction]}
>
  TestMe
</button>
`;
